lib/AST/GenericSignature.cpp

```diff
@@ -985,6 +991,7 @@ ConformanceAccessPath GenericSignature::getConformanceAccessPath(
   buildPath(getRequirements(), source, protocol, rootType, nullptr);
 
   // Return the path; we're done!
+  equivClass->conformanceAccessPathCache[protocol] = path;
```

Comment:
If we're caching these instead of building them on-demand, can we reasonably return them by reference?  Or make `ConformanceAccessPath` just hold a reference to external storage so that it's cheaper to pass one around?

Reply:
That seems sensible too.